src/main/java/lotto/domain/RankState.java

```diff
@@ -0,0 +1,59 @@
+package lotto.domain;
+
+import lotto.exception.IllegalRankStateException;
+
+import java.util.Arrays;
+
+public enum RankState {
+
+    FIRST(2000000000, 5,6),
+    SECOND(30000000, 4,5),
+    THIRD(1500000, 3,5),
+    FOURTH(50000, 2,4),
+    FIFTH(5000, 1,3),
+    FAIL(0, 0,2);
+
+    private final int winMoney;
+    private final int rankIndex;
+    private final int matchCount;
+
+    RankState(int winMoney, int rankIndex, int matchCount) {
+        this.winMoney = winMoney;
+        this.rankIndex = rankIndex;
+        this.matchCount = matchCount;
+    }
+
+    public static RankState rank(int matchCount, boolean bonusMatched) {
+        if (matchCount < FIFTH.matchCount) {
+            return FAIL;
+        }
+
+        if (SECOND.isMatchCount(matchCount)) {
+            return chooseSecondOrThird(bonusMatched);
+        }
+
+        return Arrays.stream(values())
+                .filter(rank -> rank.isMatchCount(matchCount))
+                .findFirst()
+                .orElseThrow(IllegalRankStateException::new);
```

Comment:
rank 값을 구할때 디폴트 `FAIL`을 돌려주는건 어떨까요? 그럼 0,1개일때도 자연스럽게 `FAIL`로 처리될것 같아요.
```suggestion
                return Arrays.stream(values())
                .filter(rank -> rank.isMatchCount(matchCount))
                .findFirst()
                .orElse(FAIL);

Reply:
이부분은 사실 matchCount 가 7 이상의 비정상적인 인풋을 처리하기 위함이었는데, 
말씀주신대로 변경하여 그냥 이 경우 FAIL을 반환하도록 수정하였습니다